### ISSUE TITLE
Migrating repo from NOC to EUROARGODEV

### ISSUE THREAD

**Issue by gmaze** [euroargodev/User-Acceptance-Test-Python-version-of-the-OWC-tool#10]
In order to migrate the repository from NOC to github servers at EUROARGODEV:

## Requirements:
- [x] Check if the [Github import tool](https://github.com/new/import) is effective. @gmaze tested it with success at https://github.com/gmaze/owc-software-python. All repo history and contributions are imported properly.
- [x] Check if the [CI workflow for NOC](https://git.noc.ac.uk/bodc/owc-software-python/pipelines) can be managed with github actions (or other service like [Travis](https://travis-ci.org/)). @gmaze tested with success an implementation of [NOC workflow in a Github actions](https://github.com/gmaze/owc-software-python/blob/master/.github/workflows/main.yml). Online testing is run with success ([See here for instance](https://github.com/gmaze/owc-software-python/actions/runs/115668971)) under Linux and Mac OS and with python versions 3.6, 3.7 and 3.8 (which seems to be more than what was done at NOC). However, it is unsuccessful under Windows OS, which should be fixed later.

## Migration procedure:
- [ ] Run the [Github import tool](https://github.com/new/import)
- [ ] Add a Github action workflow for CI, following @gmaze [template here](https://github.com/gmaze/owc-software-python/blob/master/.github/workflows/main.yml).
- [ ] Add code coverage (which is excellent here !) reporting
- [ ] Add badges to repo readme file

@edsmall-bodc, @matdon17 : I'll be happy to do this if you want me to.

**Comment by gmaze** [comment]
@edsmall-bodc do you confirm that the repo is located here:
https://git.noc.ac.uk/bodc/owc-software-python

the last master commit was 3 weeks ago ...

(since I have seen other links based on your NOC git fork at https://git.noc.ac.uk/edsmall)

**Comment by edsmall-bodc** [comment]
@gmaze This is correct

**Comment by gmaze** [comment]
also @matdon17 @edsmall-bodc do you agree to change the name of the repo from "owc-software-python" to "owc-software" ?
It's usually poor practice to have the language in the repo name

**Comment by edsmall-bodc** [comment]
I agree to the above. The initially naming scheme was purely because I wasn't sure if there was a company matlab version.

I realise euroargodev doesn't have the matlab version, but if the code is eventually moved to the same place as the matlab version, what title would you suggest to distinguish between the two?

**Comment by matdon17** [comment]
Agreed re: python, but isn't 'software' redundant? Maybe just owc? or argodmqc_owc?

**Comment by gmaze** [comment]
to be honest I don't know, it's just that renaming a repo is not really recommended, so we may want to have it right from the beginning. but this is not dramatic

and yes "software" is not necessary either

I like your "argodmqc_owc"
other possibilities:
pyowc
psalcalib

**Comment by gmaze** [comment]
let's go for "argodmqc_owc" ?

**Comment by edsmall-bodc** [comment]
I'd like to keep the owc to honour the original creators.

owcpy, to follow the numpy, scipy, etc convention? 

You want me to rename the gitlab version before you move it?

**Comment by gmaze** [comment]
> You want me to rename the gitlab version before you move it?
Nope

**Comment by gmaze** [comment]
almost there !

**Comment by gmaze** [comment]
Hi there,
All done !
Please have a look to the readme and tell me the access rights settings you want !

**Comment by gmaze** [comment]
Please note, that there is a possible "transfer" of issues between repositories, 

so it is feasible to migrate everything from https://github.com/euroargodev/User-Acceptance-Test-Python-version-of-the-OWC-tool to here as well. 

I think it would be more consistent to have everything in the same place, and this is still feasible since there is only 2/3 contributors to the alpha-testing

**Comment by edsmall-bodc** [comment]
Looking at the settings and README now. I agree to moving issues to be with the code.

**Comment by edsmall-bodc** [comment]
@gmaze Should I be added to the list of people highlighed here: https://github.com/orgs/euroargodev/people

I will make a new branch tomorrow, write some code, and push it, and make a merge request to check everything is working correctly, but all looks good from my side here.

In terms of access, I think at this point no merge request should be able to be merged in without

1. My consent
2. Passing all unit tests
3. Passing static code checks

Anyone should be allowed to look at and clone the code though, unless anyone has any issues with this?

I have successfully clones the repo on my machine and everything runs!

**Comment by gmaze** [comment]
> @gmaze Should I be added to the list of people highlighed here: https://github.com/orgs/euroargodev/people

I guess yes ! But this is supposedly @kamwal role to invite you to the euroargodev organisation (ask her)
In the mean time, I have invited you to argodmqc_owc and your invitation is pending ... (you'll be listed as external collaborator until @kamwal get you in)

> I will make a new branch tomorrow, write some code, and push it, and make a merge request to check everything is working correctly, but all looks good from my side here.

Good

>  In terms of access, I think at this point no merge request should be able to be merged in without:
    1. My consent
    1. Passing all unit tests
    1. Passing static code checks

All off these can be enforced in https://github.com/euroargodev/argodmqc_owc/settings/branch_protection_rules

The simplest is to list as Maintainers anyone to be authorized to push

**Comment by gmaze** [comment]
In the mean, I'm just adding a binder link to play with the repo as it is,
and then I'll move any changes to my fork

**Comment by edsmall-bodc** [comment]
At some point it would be nice to have a chat about binder - maybe when you have some time in July. Would just like to understand what it currently is, and (seeing as it's in a testing phase) what it is trying to be!

**Comment by gmaze** [comment]
yes, and it has a lot to do with the packaging of the software, pip distribution, where to put and how to handle ancillary data, etc ...

**Comment by gmaze** [comment]
I think the can close this issue when merging https://github.com/euroargodev/argodmqc_owc/pull/4

**Comment by edsmall-bodc** [comment]
I'll review and pull it in during phase 2 of testing
